### DISCUSSION TITLE
Connector: Salesforce

### DISCUSSION THREAD

**Discussion by jhaisley** [evidence-dev/evidence#2411] [Data Source Request]
## Feature Description

What would you like to see added to Evidence?
Support for salesforce as a datasource.

## Goal of Feature

What would this feature help you accomplish?
This would allow for easier adoption of evidence by schools, non-profits, & businesses which use Salesforce as their repository of truth.

## Current Solution / Workarounds

What do you do today to accomplish the goal you described above?
It's possible, but painful to export data to a supported db, then use evidence to create reports, however the data is always stale by the nature of this type of integration. 

## Alternatives

Describe any alternative solutions or features you've considered
N/A

**Comment by mcrascal** [comment]
Thanks for the suggestion! 

Would you picture querying salesforce tables using the salesforce query language, or more just listing off the salesforce tables you wanted to populate into your project?

**Comment by jhaisley** [comment]
Ideally I think they would select a table, and maybe a filter.  One big benefit would be allowing data folks who don't know SOQL but can use universal sql to create reports.

**Comment by hydrojumbo** [comment]
> One big benefit would be allowing data folks who don't know SOQL but can use universal sql to create reports.

Totally agree. For us, the Salesforce schema (and thus SOQL) change rarely. In my imagination, the SOQL would be limited to the source query, allowing reports to be composed (and data from other sources to be joined) using universal sql in evidence.

Is there a guide for contributing features such as this one? Sorry if I missed it.

**Comment by archiewood** [comment]
There is an outline of the process for creating a data source connector here:

https://docs.evidence.dev/plugins/create-source-plugin/

A template to start from is here:

https://github.com/evidence-dev/datasource-template
